linked_list/linked_list.py

```diff
@@ -86,11 +166,22 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def reverse(self):

```

Comment:
Since you have a doubly linked list you also need to think about how you redirect your previous and tail pointers.